### PR TITLE
Sto relnotes 2023 03 02 doc 2686

### PR DESCRIPTION
**Pull request by douglas-j-bothwell**
# Harness Developer Pull Request
Thanks for helping us make the Developer Hub better. The PR will be looked at
by the maintainers. 

## What Type of PR is This?

- [ ] Issue
- [ ] Feature
- [ ] Maintenance/Chore

If tied to an Issue, list the Issue(s) here:

* *Issue(s)*

## House Keeping
Some items to keep track of. Screen shots of changes are optional but would help the maintainers review quicker. 

- [ ] Tested Locally
- [ ] *Optional* Screen Shoot.

### CODE REVIEW

File: release-notes/security-testing-orchestration.md [hunk 1]
```diff
@@ -13,6 +13,38 @@ Harness deploys updates progressively to different Harness SaaS clusters. You ca
...
+### Early access features
+
+- Improved UI for configuring scan steps (STO-4867)
+  
+  This release includes a set of Security steps with an improved UI for configuring scans. Each step shows only the settings that apply to the specific scan. 
+
+  Note the following:
+
+  - This release includes new steps for the following scanners: Aqua Trivy, Bandit, Black Duck, Checkmarx, Grype, Mend, Prisma Cloud, Snyk, SonarQube, and ZAP.  
+  - These steps require a Background step running Docker-in-Docker only when scanning container images.
```
Comment (aklaft) [inline release-notes/security-testing-orchestration.md:32]: Maybe we could state this like "Docker-in-Docker no longer required except when..." 

this change is pretty big for customers and I don't want them to miss it

**Review by aklaft — APPROVED** [review]

**Review by BruceMcPhersonHarness — APPROVED** [review]

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 2]
```diff
@@ -0,0 +1,550 @@
...
+![](static/step-palette-00.png) 
+
+</details>
+
+<!-- ============================================================================= -->
+
+<details><summary>Scan settings</summary>
+
+#### Scan Mode
+
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:33]: Just going to leave the comments here for the imports so I don't lose my place.

In third bullet: `for for a scan run`

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 3]
```diff
@@ -0,0 +1,550 @@
...
+
+
+
+<details><summary>Target Settings</summary>
+
+<!-- ============================================================================= -->
+<a name="target-type"></a>
+
+#### Type
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:67]: Please add colon after each closing paren in the list (like `Repository (repositry):` or change to a complete sentence like you did in Scan Mode.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 4]
```diff
@@ -0,0 +1,550 @@
...
+import StoSettingTargetVariant from './shared/_sto-ref-ui-target-variant.md';
+```
+
+<StoSettingTargetVariant  />
+
+
+<a name="target-workspace"></a>
+
+#### Workspace (_repository_)
+
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:108]: `The default workspace path is /harness.`

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 5]
```diff
@@ -0,0 +1,550 @@
...
+```
+
+<StoSettingImageName />
+
+<!-- ============================================================================= -->
+
+
+<a name="container-tag"></a>
+
+#### Tag
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:254]: Sug to put the example in a bulleted list so they match the other ones and they aren't confused as one continuous example.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 6]
```diff
@@ -0,0 +1,550 @@
...
+<!-- ============================================================================= -->
+
+</details>
+
+<details><summary>Scan Tool Settings</summary>
+
+<!-- ============================================================================= -->
+<a name="tool-project-name"></a>
+
+#### Project Name
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:305]: Use bold for the UI names in your menu cascade. Maybe add qualifying phrase like "Go to Security Tests > Test Targets"

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 7]
```diff
@@ -0,0 +1,550 @@
...
+```
+
+<a name="product-project-version"></a>
+<StoSettingToolProjectVersion />
+
+
+<!-- ============================================================================= -->
+<a name="tool-include"></a>	
+
+#### Include
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:329]: Sug stronger wording: This must match the format required by the scanner

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 8]
```diff
@@ -0,0 +1,550 @@
...
+import StoSettingToolImageName from './shared/_sto-ref-ui-tool-context-image.md';
+```
+
+<StoSettingToolImageName />
+
+
+<!-- ============================================================================= -->
+<a name="tool-team-name"></a>
+
+#### Team Name
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:374]: Mismatch of wording: "In Checkmarx, for example, supports team names..."

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 9]
```diff
@@ -0,0 +1,550 @@
...
+```mdx-code-block
+import StoSettingTooJavaLibraries from './shared/_sto-ref-ui-tool-java-libraries.md';
+```
+
+<StoSettingTooJavaLibraries  />
+
+<!-- ============================================================================= -->
+<a name="tool-java-binaries"></a>
+
+#### Java Binaries
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:408]: Missing code syntax around sonar.java.binaries

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 10]
```diff
@@ -0,0 +1,550 @@
...
+```mdx-code-block
+import StoSettingToolProductToken from './shared/_sto-ref-ui-tool-prod-token.md';
+```
+
+<StoSettingToolProductToken  />
+
+<!-- ============================================================================= -->
+<a name="tool-product-name"></a>
+
+#### Name
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:431]: "You might want to create a secret" - sug to make "create a secret" a link, like it is in Product Token.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 11]
```diff
@@ -0,0 +1,550 @@
...
+
+</details>
+
+<details><summary>Instance settings</summary>
+
+
+<!-- ============================================================================= -->
+<a name="instance-domain"></a>
+
+#### Domain
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:467]: Use bold for the UI labels: between the **Domain** and **Path** fields. Might move the example to the first sentence so it's clear it's not an example of the path being split.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 12]
```diff
@@ -0,0 +1,550 @@
...
+```mdx-code-block
+import StoSettingInstancePort from './shared/_sto-ref-ui-instance-port.md';
+```
+
+<StoSettingInstancePort />
+
+<!-- ============================================================================= -->
+<a name="instance-path"></a>
+
+#### Path
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:499]: Path to append to the application instance domain. Only required if you're splitting the full path between the **Domain** and **Path** settings. For example, you might specify the **Domain** as https://myapp.io and the **Path** as /portal/us.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 13]
```diff
@@ -0,0 +1,550 @@
...
+<!-- ============================================================================= -->
+
+<!-- ============================================================================= -->
+
+<details><summary>Log Level, CLI flags, and Fail on Severity</summary>
+
+
+<a name="log-level"></a>
+
+#### Log Level
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:519]: The options appear to be both bold and code formatted in Log Level and Fail on Severity.

File: docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md [hunk 14]
```diff
@@ -0,0 +1,550 @@
...
+```mdx-code-block
+import StoSettingCliFlags from './shared/_sto-ref-ui-cli-flags.md';
+```
+
+<StoSettingCliFlags />
+
+<!-- ============================================================================= -->
+<a name="fail-on-severity"></a>
+
+#### Fail on severity
```
Comment (aimurphy) [inline docs/security-testing-orchestration/sto-techref-category/security-step-ui-settings-reference.md:542]: **Fail on Severity** setting.

The title has "Fail on severity" (lowercase s) but the body has "Fail on Severity" (uppercase S)

**Review by aimurphy — APPROVED** [review]
approved w assumption you'll review my comments.

**Review by StrawberryPad — APPROVED** [review]
Looks good. Deferring to April's suggestions.

**Pull request merged**